crates/miden-objects/src/batch/proposed_batch.rs

```diff
@@ -74,9 +74,10 @@ impl ProposedBatch {
     ///     - Unauthenticated note authentication can be delayed to the block kernel.
     ///     - Another transaction in the batch creates an output note matching an unauthenticated
     ///       input note, in which case inclusion in the chain does not need to be proven.
-    /// - The block header's block number must be greater or equal to the highest block number
-    ///   referenced by any transaction. This is not verified explicitly, but will implicitly cause
-    ///   an error during validating that each reference block of a transaction is in the chain MMR.
+    /// - The reference block of a batch must satisfy the following requirement: Its block number
+    ///   must be greater or equal to the highest block number referenced by any transaction. This
+    ///   is not verified explicitly, but will implicitly cause an error during the validation that
+    ///   each reference block of a transaction is in the chain MMR.
```

Comment:
@Mirko-von-Leipzig I expanded this part for what the requirement of a batch's reference block is. That should answer your first question from https://github.com/0xPolygonMiden/miden-base/pull/1168#discussion_r1963818058.

Reply:
How does this implicit checking work? Does it essentially request the chain MMR at the reference block and therefore all transactions must come before it in order to be present?